### ISSUE TITLE
Remove punctuation from names

### ISSUE THREAD

**Issue by Samalando** [PrismLauncher/PrismLauncher#2696]
Currently users have the ability to add punctuation in the instance name. I suggest that you are no longer able to do this as it causes problems when launching instances. An example is https://mclo.gs/u8UW336 and as you can see the punctuation messed launching up. I have searched the issue board and this has not been mentioned before and definitely not one that has been rejected. I am unable to use one of the formats as I'm on moblie.

**Comment by Kationor** [comment]
> @Samalando: Currently users have the ability to add punctuation in the instance name. I suggest that you are no longer able to do this as it causes problems when launching instances.

These issues are caused by the directory's path, not the instance name.

It makes more sense to sanitise the directory name while still allowing users to pick arbitrary instance names. Since this is not directly user-facing, this also allows us to be much more aggressive, as long as we keep the paths readable for the times where you *do* need to manually navigate the instances directory.

TLDR: Allow arbitrary names, sanitise paths to prevent issues.

**Comment by Samalando** [comment]
Yeah I meant like the instance folder name but it must've sounded weird. Yes I think we should be able to have names with !! And . But not the folder name
